kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/HelmMojo.java

```diff
@@ -15,81 +15,64 @@
  */
 @Mojo(name = "helm", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST)
-public class HelmMojo extends AbstractJKubeMojo {
-
-  @Component
-  MavenProjectHelper projectHelper;
+public class HelmMojo extends AbstractHelmMojo {
 
   /**
-   * The generated kubernetes YAML file
+   * The generated Kubernetes YAML file
    */
   @Parameter(property = "jkube.kubernetesManifest", defaultValue = "${basedir}/target/classes/META-INF/jkube/kubernetes.yml")
-  File kubernetesManifest;
+  private File kubernetesManifest;
 
-  /**
-   * One of:
-   * <ul>
-   *  <li>A directory containing OpenShift Templates to use as Helm parameters.</li>
-   *  <li>A file containing a Kubernetes List with OpenShift Template entries to be used as Helm parameters.</li>
-   * </ul>
-   */
-  @Parameter(property = "jkube.kubernetesTemplate", defaultValue = "${basedir}/target/classes/META-INF/jkube/kubernetes")
-  File kubernetesTemplate;
+  @Component
+  MavenProjectHelper projectHelper;
 
-  @Parameter
-  HelmConfig helm;
+  @Override
+  public void init() throws MojoFailureException {
+    super.init();
+
+    final File manifest = getKubernetesManifest();
+    if (manifest == null || !manifest.isFile()) {
+      logManifestNotFoundWarning(manifest);
+    }
+
+    final GeneratedChartListener generatedChartListener = (helmConfig, type, chartFile) -> projectHelper.attachArtifact(project, helmConfig.getChartExtension(), type.getClassifier(), chartFile);
+    getHelm().getGeneratedChartListeners().add(generatedChartListener);
+  }
 
   @Override
   public void executeInternal() throws MojoExecutionException {
     try {
-      final File manifest = getKubernetesManifest();
-      if (manifest == null || !manifest.isFile()) {
-        logManifestNotFoundWarning(manifest);
-      }
-      helm = initHelmConfig(getDefaultHelmType(), javaProject, getKubernetesTemplate(), helm)
-          .generatedChartListeners(Collections.singletonList((helmConfig, type, chartFile) -> projectHelper
-              .attachArtifact(project, helmConfig.getChartExtension(), type.getClassifier(), chartFile)))
-          .build();
-      jkubeServiceHub.getHelmService().generateHelmCharts(helm);
+      jkubeServiceHub.getHelmService().generateHelmCharts(getHelm());
     } catch (IOException exception) {
       throw new MojoExecutionException(exception.getMessage());
     }
   }
 
   protected void logManifestNotFoundWarning(File manifest) {
-    getKitLogger().warn("No kubernetes manifest file has been generated yet by the k8s:resource goal at: " + manifest);
+    getKitLogger().warn("No Kubernetes manifest file has been generated yet by the k8s:resource goal at: " + manifest);
   }
 
   protected File getKubernetesManifest() {
     return kubernetesManifest;
   }
 
-  protected File getKubernetesTemplate() {
-     return kubernetesTemplate;
-  }
-
+  @Override
   protected HelmConfig.HelmType getDefaultHelmType() {
     return HelmConfig.HelmType.KUBERNETES;
   }
-
-  HelmConfig getHelm() {
```

Comment:
Moved to [AbstractHelmMojo](https://github.com/eclipse/jkube/pull/2714/files#diff-fe068c53f13221dece005a3bd8ef7158f65321e839c8abbe12b862b2da9c3e20R59).